src/zarr/errors.py

```diff
@@ -57,3 +57,10 @@ class NodeTypeValidationError(MetadataValidationError):
     This can be raised when the value is invalid or unexpected given the context,
     for example an 'array' node when we expected a 'group'.
     """
+
+
+class RootedHierarchyError(BaseZarrError):
```

Comment:
I find this name a bit confusing, but I might not understand the context.

If we're saying "you've tried to create a root, but it already exists" then perhaps a `RootExistsError`? But I think it's actually more like, you've tried to insert a root at a level below the root? So maybe something like `RootAsChildError`, `NestedRootError`, `ChildRootError`?

Reply:
how about 
```python
class NestedRootError(BaseZarrError):
    """
    Exception raised when attempting to create a root node relative to a pre-existing root node.
    """
```

Reply:
Yeah, that sounds good.

Reply:
after a refactor, this exception isn't needed any more